### PR TITLE
Improve SGP30 humidity compensation documentation

### PR DESCRIPTION
**Pull request by aveao**
## Description:

SGP30 datasheet: https://sensirion.com/media/documents/984E0DD5/61644B8B/Sensirion_Gas_Sensors_Datasheet_SGP30.pdf

**Justification for clarification of "absolute humidity" wording and clarification that humidity source should be a relative humidity sensor:**

The datasheet makes it clear that it requires absolute humidity, which is calculated using relative humidity and temperature (celcius in the example formula):

image

The code doesn't work if it doesn't have a humidity or temperature value:

https://github.com/esphome/esphome/blob/045836c3fe4e088475c248256f7b2e537cbabe05/esphome/components/sgp30/sgp30.cpp#L165-L170

image

https://github.com/esphome/esphome/blob/045836c3fe4e088475c248256f7b2e537cbabe05/esphome/components/sgp30/sgp30.cpp#L174-L180

image

**Justification for clarification of celcius requirement:**

Code uses same algorithm in example and also explicitly prints the log entries in celcius:

https://github.com/esphome/esphome/blob/045836c3fe4e088475c248256f7b2e537cbabe05/esphome/components/sgp30/sgp30.cpp#L182

**Related issue (if applicable):** not applicable

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** not applicable

## Checklist:

  - [ ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [x] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by aveao** [comment]
I hope a message is enough to satisfy stalebot:

I generally don't find the use of stalebot tasteful, especially if it cannot handle the nuance of "I have been waiting for a review for months". Not reviewing my contribution while it is subject to a potential auto-closure makes it feel like contributions are not valued here. This goes for any repo that uses such arrangements.

edit: apparently not even a rebase and a comment will satisfy stalebot.

**Comment by aveao** [comment]
@jesserockz @OttoWinter @nagyrobi
Hi, I'm sorry for pinging you all here but the README doesn't link to anything like a development channel where I can reach people, and you three are the people that contributed most to this repo ([with Otto also introducing this version of stalebot](https://github.com/esphome/esphome-docs/pull/1566)) who are also part of the esphome org.

tl;dr: Using a stalebot is really discouraging as a contributor.

With this PR I took time of my day to not only figure things out for myself, but then assemble the information and then put it into a nice shape with which it can hopefully be merged into the repo, and write a good long description to help explain all the changes, to make things better for everyone.
For that I am of course not owed a timely review, but having my PR be at risk of closure because I've done my best but you haven't found time to review the PR in a timely fashion is simply insulting. The widespreadness of this type of thing makes me reconsider if I want to contribute to OSS projects at all.

Please reconsider using stalebot, especially in a config that doesn't let me bump to confirm that the PR is still valid. Or alternatively pair it with a bot to automatically assign a reviewer. Until this situation is improved, I won't be contributing to esphome (and in general, I'll avoid contributing to projects with stalebots in the future).

File: components/sensor/sgp30.rst [hunk 1]
```diff
@@ -72,12 +72,12 @@ Advanced:
   - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
   - All other options from :ref:`Sensor <config-sensor>`.
 
-- **compensation** (*Optional*): The block containing sensors used for compensation.
+- **compensation** (*Optional*): The block containing sensors used for compensation. Both values should be supplied in order to be able to generate the absolute humidity to be reported to the sensor.
 
   - **temperature_source** (*Optional*, :ref:`config-id`): Give an external temperature sensor ID
-    here. This can improve the sensor's internal calculations.
+    here. The data should be in Celcius. This can improve the sensor's internal calculations.
```
Comment (guillempages) [inline components/sensor/sgp30.rst]: Cel*s*ius
```suggestion
    here. The data should be in Celsius. This can improve the sensor's internal calculations.
```
Also maybe explain why? To me "should" means that it would still work in other units, so what is the advantage of using Celsius? Or is this actually a "must" to get meaningful values at all?

Comment (aveao) [reply components/sensor/sgp30.rst]: Does this repo follow RFC2119? I haven't seen any mentions of that anywhere, I'm fairly sure I could find other instances of it being violated if I tried. (in fact: [in the linter](https://github.com/esphome/esphome-docs/blob/1bdfb560a6572742399175976a8c629755c49f4e/lint.py#L283), [in a component](https://github.com/esphome/esphome-docs/blob/1bdfb560a6572742399175976a8c629755c49f4e/components/sn74hc595.rst#L15), [in a sensor](https://github.com/esphome/esphome-docs/blob/1bdfb560a6572742399175976a8c629755c49f4e/components/sensor/tx20.rst#L14), prob many more around)

Regardless, yes, it's a "must": See my explanation above in "Justification for clarification of celcius requirement".

I'll correct the wording.

**Comment by nagyrobi** [comment]
I don't have such a sensor to confirm myseld.

**Comment by aveao** [comment]
> I don't have such a sensor to confirm myseld.

I tried to clarify all changes in the OP text, no physical sensor is needed, datasheet and the code makes it clear that the changes are correct.

Anyhow, I tagged you to state my concerns about the use of stalebot in PRs, not to demand a review from you or anything. Sorry if this wasn't clera.

**Comment by aveao** [comment]
PR is still valid.

**Review by nagyrobi — APPROVED** [review]
